lib/Config/Menu.php

```diff
@@ -836,6 +837,13 @@ function checkPremiumKey(ServicesChecker $checker = null) {
     $this->premium_key_valid = $checker->isPremiumKeyValid($show_notices);
   }
 
+  function checkFromEmailAuthorization(ServicesChecker $checker = null) {
+    if (self::isOnMailPoetAdminPage()) {
+      $checker = $checker ?: new ServicesChecker();
```

Comment:
Can we pass `ServicesChecker` through DI instead creating it here?

Reply:
Should I change it everywhere in this file? It's used in three more functions.

Reply:
Yes, please.